pallets/loan/src/math.rs

```diff
@@ -125,6 +133,29 @@ where
 		.and_then(|d| expected_cash_flow.checked_div(&d))
 }
 
+/// returns the valid write off group give the maturity date and current time
+/// since the write off groups are not guaranteed to be in a sorted order and
+/// we want to preserve the index of the group,
+/// we also pick the group that has the highest overdue days found in the vector
+pub(crate) fn valid_write_off_group<Rate>(
```

Comment:
A suggestion here would be to first sort the vec by `overdue_days` DESC and then find the first with a valid offset. That would remove all the procedural and mutable code. Leave it up to you!

Reply:
we can't sort them, then out indexs will change and we will end picking the wrong write off group.